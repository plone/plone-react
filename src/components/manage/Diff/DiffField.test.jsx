import React from 'react';
// import renderer from 'react-test-renderer';
import { waitFor, render, screen } from '@testing-library/react';

import DiffField from './DiffField';

jest.mock('moment', () =>
  jest.fn(() => ({
    format: jest.fn(() => 'Sunday, April 23, 2017 3:38 AM'),
  })),
);

describe('DiffField', () => {
  it('renders a diff field in split mode', async () => {
    const component = render(
      <DiffField
        pathname="/blog"
        schema={{ title: 'Title', type: 'string' }}
        one="My old string"
        two="My new string"
        view="split"
      />,
    );
    await waitFor(() => screen.getByTestId('DiffField'));
    expect(component).toMatchSnapshot();
    // const json = component.toJSON();
    // expect(json).toMatchSnapshot();
  });

  it('renders a diff field in unified mode', async () => {
    const component = render(
      <DiffField
        pathname="/blog"
        schema={{ title: 'Title', type: 'string' }}
        one="My old string"
        two="My new string"
        view="unified"
      />,
    );
    await waitFor(() => screen.getByTestId('DiffField'));
    expect(component).toMatchSnapshot();
  });

  it('renders a richtext field', async () => {
    const component = render(
      <DiffField
        pathname="/blog"
        schema={{ widget: 'richtext', title: 'Text', type: 'string' }}
        one={{ data: 'My old string' }}
        two={{ data: 'My new string' }}
        view="unified"
      />,
    );
    await waitFor(() => screen.getByTestId('DiffField'));
    expect(component).toMatchSnapshot();
  });

  it('renders a datetime field', async () => {
    const component = render(
      <DiffField
        pathname="/blog"
        schema={{ widget: 'datetime', title: 'Text', type: 'string' }}
        one="2017-04-25T16:14:18+02:00"
        two="2016-04-25T16:14:18+02:00"
        view="unified"
      />,
    );
    await waitFor(() => screen.getByTestId('DiffField'));
    expect(component).toMatchSnapshot();
  });

  it('renders a textarea field', async () => {
    const component = render(
      <DiffField
        pathname="/blog"
        schema={{ widget: 'textarea', title: 'Text', type: 'string' }}
        one="My old string"
        two="My new string"
        view="unified"
      />,
    );
    await waitFor(() => screen.getByTestId('DiffField'));
    expect(component).toMatchSnapshot();
  });

  it('renders a boolean field', async () => {
    const component = render(
      <DiffField
        pathname="/blog"
        schema={{ title: 'Text', type: 'boolean' }}
        one
        two={false}
        view="unified"
      />,
    );
    await waitFor(() => screen.getByTestId('DiffField'));
    expect(component).toMatchSnapshot();
  });

  it('renders a boolean field with inverse values', async () => {
    const component = render(
      <DiffField
        pathname="/blog"
        schema={{ title: 'Text', type: 'boolean' }}
        one={false}
        two
        view="unified"
      />,
    );
    await waitFor(() => screen.getByTestId('DiffField'));
    expect(component).toMatchSnapshot();
  });

  it('renders an array field', async () => {
    const component = render(
      <DiffField
        pathname="/blog"
        schema={{ title: 'Text', type: 'array' }}
        one={['one', 'two']}
        two={['one', 'three']}
        view="unified"
      />,
    );
    await waitFor(() => screen.getByTestId('DiffField'));
    expect(component).toMatchSnapshot();
  });

  it('renders an array of objects field', async () => {
    const component = render(
      <DiffField
        pathname="/blog"
        schema={{ title: 'Text', type: 'array' }}
        one={[
          { title: 'one', token: 'one' },
          { title: 'two', token: 'two' },
        ]}
        two={[
          { title: 'one', token: 'one' },
          { title: 'three', token: 'three' },
        ]}
        view="unified"
      />,
    );
    await waitFor(() => screen.getByTestId('DiffField'));
    expect(component).toMatchSnapshot();
  });
});
