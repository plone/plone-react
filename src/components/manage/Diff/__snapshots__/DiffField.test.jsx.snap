// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DiffField renders a boolean field 1`] = `
<div>
  <table
    class="ui compact table"
    data-testid="DiffField"
  >
    <thead
      class=""
    >
      <tr
        class=""
      >
        <th
          class=""
          colspan="2"
        >
          Text
        </th>
      </tr>
    </thead>
    <tbody
      class=""
    >
      <tr
        class=""
      >
        <td
          class="top aligned sixteen wide"
        >
          <span>
            <span
              class="deletion"
            >
              true
            </span>
            <span
              class="addition"
            >
              false
            </span>
          </span>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`DiffField renders a boolean field with inverse values 1`] = `
<div>
  <table
    class="ui compact table"
    data-testid="DiffField"
  >
    <thead
      class=""
    >
      <tr
        class=""
      >
        <th
          class=""
          colspan="2"
        >
          Text
        </th>
      </tr>
    </thead>
    <tbody
      class=""
    >
      <tr
        class=""
      >
        <td
          class="top aligned sixteen wide"
        >
          <span>
            <span
              class="deletion"
            >
              false
            </span>
            <span
              class="addition"
            >
              true
            </span>
          </span>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`DiffField renders a datetime field 1`] = `
<div>
  <table
    class="ui compact table"
    data-testid="DiffField"
  >
    <thead
      class=""
    >
      <tr
        class=""
      >
        <th
          class=""
          colspan="2"
        >
          Text
        </th>
      </tr>
    </thead>
    <tbody
      class=""
    >
      <tr
        class=""
      >
        <td
          class="top aligned sixteen wide"
        >
          <span>
            <span>
              Sunday, April 23, 2017 3:38 AM
            </span>
          </span>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`DiffField renders a diff field in split mode 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <table
        class="ui compact table"
        data-testid="DiffField"
      >
        <thead
          class=""
        >
          <tr
            class=""
          >
            <th
              class=""
              colspan="2"
            >
              Title
            </th>
          </tr>
        </thead>
        <tbody
          class=""
        >
          <tr
            class=""
          >
            <td
              class="top aligned eight wide"
            >
              <span>
                <span>
                  My 
                </span>
                <span
                  class="deletion"
                >
                  old
                </span>
                <span>
                   string
                </span>
              </span>
            </td>
            <td
              class="top aligned eight wide"
            >
              <span>
                <span>
                  My 
                </span>
                <span
                  class="addition"
                >
                  new
                </span>
                <span>
                   string
                </span>
              </span>
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </body>,
  "container": <div>
    <table
      class="ui compact table"
      data-testid="DiffField"
    >
      <thead
        class=""
      >
        <tr
          class=""
        >
          <th
            class=""
            colspan="2"
          >
            Title
          </th>
        </tr>
      </thead>
      <tbody
        class=""
      >
        <tr
          class=""
        >
          <td
            class="top aligned eight wide"
          >
            <span>
              <span>
                My 
              </span>
              <span
                class="deletion"
              >
                old
              </span>
              <span>
                 string
              </span>
            </span>
          </td>
          <td
            class="top aligned eight wide"
          >
            <span>
              <span>
                My 
              </span>
              <span
                class="addition"
              >
                new
              </span>
              <span>
                 string
              </span>
            </span>
          </td>
        </tr>
      </tbody>
    </table>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`DiffField renders a diff field in unified mode 1`] = `
<div>
  <table
    class="ui compact table"
    data-testid="DiffField"
  >
    <thead
      class=""
    >
      <tr
        class=""
      >
        <th
          class=""
          colspan="2"
        >
          Title
        </th>
      </tr>
    </thead>
    <tbody
      class=""
    >
      <tr
        class=""
      >
        <td
          class="top aligned sixteen wide"
        >
          <span>
            <span>
              My 
            </span>
            <span
              class="deletion"
            >
              old
            </span>
            <span
              class="addition"
            >
              new
            </span>
            <span>
               string
            </span>
          </span>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`DiffField renders a richtext field 1`] = `
<div>
  <table
    class="ui compact table"
    data-testid="DiffField"
  >
    <thead
      class=""
    >
      <tr
        class=""
      >
        <th
          class=""
          colspan="2"
        >
          Text
        </th>
      </tr>
    </thead>
    <tbody
      class=""
    >
      <tr
        class=""
      >
        <td
          class="top aligned sixteen wide"
        >
          <span>
            <span>
              My 
            </span>
            <span
              class="deletion"
            >
              old
            </span>
            <span
              class="addition"
            >
              new
            </span>
            <span>
               string
            </span>
          </span>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`DiffField renders a textarea field 1`] = `
<div>
  <table
    class="ui compact table"
    data-testid="DiffField"
  >
    <thead
      class=""
    >
      <tr
        class=""
      >
        <th
          class=""
          colspan="2"
        >
          Text
        </th>
      </tr>
    </thead>
    <tbody
      class=""
    >
      <tr
        class=""
      >
        <td
          class="top aligned sixteen wide"
        >
          <span>
            <span>
              My 
            </span>
            <span
              class="deletion"
            >
              old
            </span>
            <span
              class="addition"
            >
              new
            </span>
            <span>
               string
            </span>
          </span>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`DiffField renders an array field 1`] = `
<div>
  <table
    class="ui compact table"
    data-testid="DiffField"
  >
    <thead
      class=""
    >
      <tr
        class=""
      >
        <th
          class=""
          colspan="2"
        >
          Text
        </th>
      </tr>
    </thead>
    <tbody
      class=""
    >
      <tr
        class=""
      >
        <td
          class="top aligned sixteen wide"
        >
          <span>
            <span>
              one,
            </span>
            <span
              class="deletion"
            >
              two
            </span>
            <span
              class="addition"
            >
              three
            </span>
          </span>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`DiffField renders an array of objects field 1`] = `
<div>
  <table
    class="ui compact table"
    data-testid="DiffField"
  >
    <thead
      class=""
    >
      <tr
        class=""
      >
        <th
          class=""
          colspan="2"
        >
          Text
        </th>
      </tr>
    </thead>
    <tbody
      class=""
    >
      <tr
        class=""
      >
        <td
          class="top aligned sixteen wide"
        >
          <span>
            <span>
              one,
            </span>
            <span
              class="deletion"
            >
              two
            </span>
            <span
              class="addition"
            >
              three
            </span>
          </span>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
